notebooks/allen_smFISH.py

```diff
@@ -0,0 +1,201 @@
+# EPY: END markdown
+
+# EPY: START code
+from starfish.pipeline.filter.clip import Clip
+s_clip = Clip(p_min=10, p_max=100, verbose=True)
+s_clip.filter(s)
+# EPY: END code
+
+# EPY: START markdown
+# We're still working through the backing of the Stack.image object with the on-disk or on-cloud Tile spec. As a result, most of our methods work in-place. For now, we can hack around this by deepcopying the data before administering the operation. I'm doing this on a workstation, so be aware of the memory usage!
+# EPY: END markdown
+
+# EPY: START code
+# filtered_backup = deepcopy(s)
+# EPY: END code
+
+# EPY: START markdown
+# If you ever want to visualize the image in the notebook, we've added a widget to do that. The first parameter is an indices dict that specifies which hybridization round, channel, z-slice you want to view. The result is a pageable visualization across that arbitrary set of slices. Below I'm visualizing the first channel, which your codebook tells me is Nmnt. 
+# 
+# [N.B. once you click on the slider, you can page with the arrow keys on the keyboard.]
+# EPY: END markdown
+
+# EPY: START code
+s.image.show_stack({'c': 0});
+# EPY: END code
+
+# EPY: START code
+from starfish.pipeline.filter.bandpass import Bandpass
+s_bandpass = Bandpass(lshort=0.5, llong=7, threshold=None, truncate=4, verbose=True)
+s_bandpass.filter(s)
+# EPY: END code
+
+# EPY: START markdown
+# For bandpass, there's a point where things get weird, at `c == 0; z <= 14`. In that range the images look mostly like noise. However, _above_ that, they look great + background subtracted! The later stages of the pipeline appear robust to this, though, as no spots are called for the noisy sections. 
+# EPY: END markdown
+
+# EPY: START code
+# I wasn't sure if this clipping was supposed to be by volume or tile. I've done tile here, but it can be easily
+# switched to volume. 
+s_clip = Clip(p_min=10, p_max=100, is_volume=False, verbose=True)
+s_clip.filter(s)
+# EPY: END code
+
+# EPY: START code
+from starfish.pipeline.filter.gaussian_low_pass import GaussianLowPass
+
+sigma=(1, 0, 0)  # filter only in z, do nothing in x, y
+glp = GaussianLowPass(sigma=sigma, is_volume=True, verbose=True)
+glp.filter(s)
+# EPY: END code
+
+# EPY: START markdown
+# Below, because spot finding is so slow when single-plex, we'll pilot this on a max projection to show that the parameters work. Here's what trackpy.locate, which we wrap, produces for a z-projection of channel 1. To do use our plotting methods on z-projections we have to expose some of the starfish internals, which will be improved upon. 
+# EPY: END markdown
+
+# EPY: START code
+from showit import image
+from trackpy import locate
+
+# grab a section from the tensor. 
+ch1 = s.max_proj(('z'))[0, 1]
```

Comment:
Is it intentional that you're not using `Indices`?

Reply:
In the sense that I'm being stupid because I find it more intuitive to use the string when I'm writing, I guess. 

But you should keep reminding me -- much better to use Indices :)